src/main/java/com/redhat/exhort/providers/GoModulesProvider.java

```diff
@@ -152,47 +156,45 @@ private void performManifestVersionsCheck(String[] goModGraphLines, Path manifes
-                          String currentVersion = artifactParts[1];
-                          if (currentDepName.trim().equals(depName.trim())) {
-                            if (!currentVersion.trim().equals(version.trim())) {
-                              throw new RuntimeException(
-                                  String.format(
-                                      "Can't continue with analysis - versions mismatch for"
-                                          + " dependency name=%s, manifest version=%s, installed"
-                                          + " Version=%s, if you want to allow version mismatch for"
-                                          + " analysis between installed and requested packages,"
-                                          + " set environment variable/setting -"
-                                          + " %s=false",
-                                      depName,
-                                      currentVersion,
-                                      version,
-                                      Provider.PROP_MATCH_MANIFEST_VERSIONS));
-                            }
-                          }
-                        });
-              });
+      comparisonLines.forEach(
+          (dependency) -> {
+            String[] parts = dependency.split("@");
+            String version = parts[1];
+            String depName = parts[0];
+            goModDependencies.forEach(
+                (dep) -> {
+                  String[] artifactParts = dep.trim().split(" ");
+                  String currentDepName = artifactParts[0];
+                  String currentVersion = artifactParts[1];
+                  if (currentDepName.trim().equals(depName.trim())) {
+                    if (!currentVersion.trim().equals(version.trim())) {
+                      throw new RuntimeException(
+                          String.format(
+                              "Can't continue with analysis - versions mismatch for"
+                                  + " dependency name=%s, manifest version=%s, installed"
+                                  + " Version=%s, if you want to allow version mismatch for"
+                                  + " analysis between installed and requested packages,"
+                                  + " set environment variable/setting -"
+                                  + " %s=false",
+                              depName,
+                              currentVersion,
+                              version,
+                              Provider.PROP_MATCH_MANIFEST_VERSIONS));
+                    }
+                  }
+                });
+          });
     } catch (IOException e) {
       throw new RuntimeException(
           "Failed to open go.mod file for manifest versions check validation!");
     }
   }
 
   private List<String> collectAllDepsFromManifest(String[] lines, String goModLines) {
-    List<String> result = new ArrayList<>();
+    List<String> result;
```

Comment:
you can also assign the value directly in the delaration
```suggestion
    List<String> result =
        Arrays.stream(lines)
            .filter((line) -> line.trim().startsWith("require") && !line.contains("("))
            .map((dep) -> dep.substring("require".length()).trim())
            .collect(Collectors.toList());
```

Reply:
updated. thanks.